kyo-prelude/shared/src/main/scala/kyo2/kernel/Loop.scala

```diff
@@ -7,212 +7,239 @@ import scala.util.NotGiven
 
 object Loop:
 
-    case class Continue[Input](input: Input)
-    private case class Continue2[Input1, Input2](input1: Input1, input2: Input2)
-    private case class Continue3[Input1, Input2, Input3](input1: Input1, input2: Input2, input3: Input3)
+    abstract class Continue[V1]:
```

Comment:
How about `A, B, C`, etc for type parameters? You can keep `O` distinct if you want.